### PR TITLE
WARP-729: Android rotate fixes

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Problem
In the Android mobile app, rotating the device in certain scenarios resulted in the rendered content to not resize, resize improperly, and disappear.

## Solution
* Perform null checking in `ZoomPersistenceRegistrar` (fixes disappearing content).
* Change `EventSynthesizer._resizeEventType` to always be `"resize"`, even for native mobile apps (fixes improperly-resizing content).

#### Android
* Download [a pre-built .apk I've shared on Google Drive](https://drive.google.com/a/webfilings.com/file/d/0B0HjwL5U4UfLdWpOcHdiWFJ5OE0/view?usp=sharing) and install it.
* OR, build it:
    * Check out wdesk-android branch [WARP-729-test](https://github.com/toddtarbox-wf/wdesk-android/compare/WARP-729-test).
    * Run `grunt`.
    * Build the app in Android Studio with the `betaDebug` variant.
* Launch the app.
* In a multi-page binder (Books viewer):
    *  Zoom in to a page all the way, rotate the device, and verify that:
        * the content resizes properly ([image](https://jira.webfilings.com/secure/attachment/253403/Android%20Rotation%201.png) of improper resizing).
        * there is no "split-view" effect that displays and cuts off two pages ([image](https://jira.webfilings.com/secure/attachment/253404/Android%20Rotation%202.png)).
    *  Rotate a whole bunch and verify that the content resizes properly.
        * _Note: there might be a slight delay (~500ms) in content resizing after rotating, but that can potentially be lowered once the `EventSynthesizer` changes go through._
    *  Zoom out from a page all the way, rotate the device, and verify that the content resizes properly and is displayed (as opposed to missing, leaving you with a grey background).
        * __Note:__ this may only occur on Android Lollipop, so be sure to verify this step on that OS.
* In a multi-page document (Swift viewer), perform general regression testing around rotation.

#### iOS
iOS didn't seem to be affected by these issues, but we should test for regressions:
* Check out wdesk-ios branch [WARP-729-test](https://github.com/bryanhales-wf/wdesk-ios/compare/WARP-729-test).
* Run `grunt`.
* Build the app in Xcode and launch it.
* Test for regressions when rotating in both binders and documents.

## Unit Tests
Added

---
@timmccall-wf @georgelesica-wf @clairesarsam-wf @toddtarbox-wf @bryanhales-wf @brianreed-wf @brentarndorfer-wf

### CODE REVIEW

**Comment by rmconsole-wf** [comment]
When this pull is merged I will not be able to function 
		properly because no valid fixversions exist for 
		Project: WARP
		Repo: Workiva/wf-uicomponents

***
*Last updated on Friday, January 23 07:51 PM CST*

**Comment by timmccall-wf** [comment]
+1 

Thanks!

File: src/awesome_map/EventSynthesizer.js [hunk 1]
```diff
@@ -171,11 +171,11 @@ define(function(require) {
...
 
         /**
-         * Depending on environment, trigger resize on 'resize' or 'orientationChange'
-         * @type {Boolean}
+         * Trigger resize on 'resize'.
+         * @type {string}
          * @private
          */
-        this._resizeEventType = (typeof dependencies.getWindow().cordova !== 'undefined') ? 'orientationchange' : 'resize';
+        this._resizeEventType = 'resize';
```
Comment (timmccall-wf) [inline src/awesome_map/EventSynthesizer.js]: Not sure if we event need this. Could just inline the string as it's now constant.

File: src/scroll_list/ZoomPersistenceRegistrar.js [hunk 2]
```diff
@@ -24,7 +24,11 @@ define(function() {
             if (currentItemIndex !== scrollList.getCurrentItem().index) {
                 return;
             }
-            var newScale = scrollList.getCurrentItemMap().getScale();
+            var currentItemMap = scrollList.getCurrentItemMap();
+            if (! currentItemMap) {
```
Comment (timmccall-wf) [inline src/scroll_list/ZoomPersistenceRegistrar.js]: Unnecessary whitespace

**Comment by timmccall-wf** [comment]
+1

Minor comments, though. Thanks!

File: src/awesome_map/EventSynthesizer.js (hunk 1 above)
Comment (greglittlefield-wf) [inline src/awesome_map/EventSynthesizer.js]: Good point; I'll inline it.

File: src/scroll_list/ZoomPersistenceRegistrar.js (hunk 2 above)
Comment (greglittlefield-wf) [inline src/scroll_list/ZoomPersistenceRegistrar.js]: Ack, sorry 'bout that. Will remove.

**Comment by toddtarbox-wf** [comment]
+10 - tested on N7 with 5.0.1, all previous rotation issues were resolved.

**Comment by timmccall-wf** [comment]
I'd like to hold off on merge until iOS rotation is +10'd as well

**Comment by charliekump-wf** [comment]
@rmconsole-wf

**Comment by timmccall-wf** [comment]
+10 for in-browser resize handling

**Comment by bryanhales-wf** [comment]
I don't see any negative consequences on iOS. 

**+10 from the iOS perspective**

**Comment by robertharwell-wf** [comment]
Binders QA +1

Tested on Nexus 10, Nexus 5, and iOS sim.  Rotation does not reproduce the canvas redraw issue described in this ticket.  Further, zoom in and rotate behaves as expected.  No regressions.

A issue was discovered that is unrelated in WARP-759.  It pertains to the width of the display area on phones during a rotate, but issue lives in the native code.

JIRA Ticket:  https://jira.webfilings.com/browse/WARP-729
Code Review PR:  https://github.com/Workiva/wf-uicomponents/pull/130
Unit Tests:  Yes
Passing Smithy:  https://travis-ci.org/Workiva/wf-uicomponents/builds/48311776

**Comment by robertharwell-wf** [comment]
@charliekump-wf - Please review and merge as needed.

**Comment by charliekump-wf** [comment]
QA Resource Approval: +1

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Unit test created/updated  
- [x] All unit tests pass 

Merging into master.
